receiver.go

```diff
@@ -139,25 +138,7 @@ func (r *receiveHandler) receiveResponse(rpc *rpcResponseMessage) (err error) {
 	return nil
 }
 
-func (r *receiveHandler) Close(err error) chan struct{} {
+func (r *receiveHandler) Close() {
 	close(r.stopCh)
-	r.broadcast(err)
-	return r.closedCh
-}
-
-func (r *receiveHandler) AddCloseListener(ch chan<- error) {
-	r.listenerMtx.Lock()
-	defer r.listenerMtx.Unlock()
-	r.listeners[ch] = struct{}{}
-}
-
-func (r *receiveHandler) broadcast(err error) {
-	r.listenerMtx.Lock()
-	defer r.listenerMtx.Unlock()
-	for ch := range r.listeners {
-		select {
-		case ch <- err:
-		default:
-		}
-	}
+	<-r.closedCh
```

Comment:
Also I'd rather return `r.closedCh` rather than blocking on it in this method, giving the caller the option to block or not.